behavioral/iterator_test.go

```diff
@@ -11,7 +11,7 @@ func TestIterator(t *testing.T) {
 
 	iterator := ArrayIterator{array, 0}
```

Comment:
Thanks for the PR and good catch. Probably better to put the ampersand here:

	iterator := &ArrayIterator{array, 0}

Reply:
Could you also fix the typo in the commit comment please? I.e. address.